conbench/entities/commit.py

```diff
@@ -424,25 +422,30 @@ def get_commits_to_branch(
         """Get information about each commit on a given branch.
 
         since and until are inclusive.
+
+        Expect tz-naive datetime objects, or expect tz-aware objects with UTC
+        timezone.
         """
         if name == "org/repo":
             # test case
             return []
 
         if ":" in branch:
             branch = branch.split(":")[1]
-        since = since.replace(tzinfo=None).isoformat() + "Z"
-        until = until.replace(tzinfo=None).isoformat() + "Z"
+
+        since_iso_for_url = since.replace(tzinfo=None).isoformat() + "Z"
+        until_iso_for_url = until.replace(tzinfo=None).isoformat() + "Z"
+        del since, until
 
         log.info(
-            f"Finding all commits to the {branch} branch of {name} between {since} and "
-            f"{until}"
+            f"Finding all commits to the {branch} branch of {name} between "
+            f" {since_iso_for_url} and {until_iso_for_url}"
         )
         url = (
             f"{GITHUB}/repos/{name}/commits?per_page=100&sha={branch}"
-            f"&since={since}&until={until}"
+            f"&since={since_iso_for_url}&until={until_iso_for_url}"
         )
-        commits = []
+        commits: List[Dict] = []
```

Comment:
Using 
```python
class CommitDict(TypedDict):
   sha: str
   repository: str
   github: str
```
this can become
```py
List[CommitDict]
```

Reply:
Sure! Love `TypeDict`. Will however punt on this. This specific change was to quickly eradicate an error.

Adding useful types for our custom primary data structures around Run, Benchmark, BenchmarkResult, Commit, etc comes later.

I tried adding useful types in this PR, and quickly realized that I have to down-scope this.